checkout/interfaces/controller/checkoutcontroller.go

```diff
@@ -546,13 +549,22 @@ func (cc *CheckoutController) ReviewAction(ctx context.Context, r *web.Request)
 		DecoratedCart: *decoratedCart,
 	}
 
+	flashErrors := r.Session().Flashes(CheckoutErrorFlashKey)
```

Comment:
You use nearly the same code block in line 380ff except the return, does it make sense to build a private function to avoid code duplication?

Reply:
Moved logic to a helper func.